core/string/ustring.cpp

```diff
@@ -4770,8 +4823,9 @@ String String::sprintf(const Array &values, bool *error) const {
 						return "not enough arguments for format string";
 					}
 
-					if (!values[value_index].is_num()) {
-						return "* wants number";
+					Variant::Type valueType = values[value_index].get_type();
+					if (!values[value_index].is_num() && valueType != Variant::VECTOR2 && valueType != Variant::VECTOR3 && valueType != Variant::VECTOR4 && valueType != Variant::VECTOR2I && valueType != Variant::VECTOR3I && valueType != Variant::VECTOR4I) {
```

Comment:
```suggestion
					if (!values[value_index].is_num()
							&& valueType != Variant::VECTOR2 && valueType != Variant::VECTOR2I
							&& valueType != Variant::VECTOR3 && valueType != Variant::VECTOR3I
							&& valueType != Variant::VECTOR4 && valueType != Variant::VECTOR4I) {
```
a bit more readable.

Reply:
Sure